datacube/storage/_read.py

```diff
@@ -129,6 +129,7 @@ def read_time_slice(rdr,
 
     is_nn = is_resampling_nn(resampling)
     scale = pick_read_scale(rr.scale, rdr)
+    scale_xy = [pick_read_scale(s, rdr) for s in rr.scale2]
```

Comment:
Not sure if `pick_read_scale` is needed here, certainly no need to pass through `rdr` object, right now it's no-op regardless but the idea is to pick an overview image best fitting current scale.
```suggestion
    xscale, yscale = [pick_read_scale(s) for s in rr.scale2]
```

Reply:
Yeah, it's mostly just force it to be an integer.